### ISSUE TITLE
UNASSIGNED-BestPractices-SemaphoreCount and similar are spammy and not useful

### ISSUE THREAD

**Issue by ShabbyX** [KhronosGroup/Vulkan-ValidationLayers#4654]
**TL;DR please remove these messages**

A class of best practices messages like `UNASSIGNED-BestPractices-SemaphoreCount` say something like "count is 0, but pointer is not NULL". This message doesn't seem useful; it's pretty common to have code like this, and it's perfectly legitimate code:

```
data.pData = dataStorage;
data.dataCount = count;
```

To silence this message, the app would have to do this:

```
data.pData = count == 0 ? NULL : dataStorage;
data.dataCount = count;
```

which is arguably not an improvement. However, this by itself is not a big deal, just a minor inconvenience.

The reason I brought this up is this however: https://github.com/baldurk/renderdoc/pull/2739

When running an app that enables best practices on RenderDoc, due to RenderDoc's capture and reconstruction of the apps structs, this message spams the error output in RenderDoc even if the app doesn't originally trigger it. While this could be worked around in RenderDoc, it may be better to just stop spamming apps with this message.
